lua/lualine/components/branch/git_branch.lua

```diff
@@ -62,42 +62,48 @@ end
 function M.find_git_dir(dir_path)
   -- get file dir so we can search from that dir
   local file_dir = dir_path or vim.fn.expand('%:p:h')
-  local root_dir = file_dir
   local git_dir
-  -- Search upward for .git file or folder
-  while root_dir do
-    if git_dir_cache[root_dir] then
-      git_dir = git_dir_cache[root_dir]
-      break
-    end
-    local git_path = root_dir .. sep .. '.git'
-    local git_file_stat = vim.loop.fs_stat(git_path)
-    if git_file_stat then
-      if git_file_stat.type == 'directory' then
-        git_dir = git_path
-      elseif git_file_stat.type == 'file' then
-        -- separate git-dir or submodule is used
-        local file = io.open(git_path)
-        if file then
-          git_dir = file:read()
-          git_dir = git_dir and git_dir:match('gitdir: (.+)$')
-          file:close()
-        end
-        -- submodule / relative file path
-        if git_dir and git_dir:sub(1, 1) ~= sep and not git_dir:match('^%a:.*$') then
-          git_dir = git_path:match('(.*).git') .. git_dir
-        end
+  if vim.env.GIT_DIR then
+    git_dir = vim.env.GIT_DIR
+    file_dir = git_dir
```

Comment:
why are you setting file_dir to git_dir ?? this is just used to cache git directory for the folder current file is in so we don't make the search every time git component runs.

also instead of putting main search logic in an else block. 

check if GIT_DIR is set. use that value to check if we need to update state. then early return from the function once the proceeding is done.

something like this
```lua
 if vim.env.GIT_DIR then
    git_dir = vim.env.GIT_DIR
    -- you can probably refactor this logic into a separate function so code repetition isn't needed
    git_dir_cache[file_dir] = git_dir
    if dir_path == nil and current_git_dir ~= git_dir then
      current_git_dir = git_dir
      update_branch()
    end
    return git_dir
  end
    
```

Reply:
I've made the change as you suggested, and also removed the `git_dir_cache` assignment, since it wouldn't be used anyway when `GIT_DIR` is set. Let me know what you think.